src/Migrate_ast.ml

```diff
@@ -40,6 +42,23 @@ module Printast = struct
           PPat
             ( to_current.copy_pattern x
             , Option.map ~f:to_current.copy_expression y ) )
+
+  let use_file f (x: Parsetree.toplevel_phrase list) =
+    List.iter x ~f:(fun x ->
+        top_phrase f
+          ( match (x : Parsetree.toplevel_phrase) with
+          | Parsetree.Ptop_def x -> Ptop_def (to_current.copy_structure x)
+          | Ptop_dir (a, b) ->
+              let open Parsetree in
+              let b =
+                match b with
+                | Pdir_none -> Pdir_none
+                | Pdir_string s -> Pdir_string s
+                | Pdir_int (s, c) -> Pdir_int (s, c)
+                | Pdir_ident i -> Pdir_ident i
+                | Pdir_bool b -> Pdir_bool b
+              in
+              Ptop_dir (a, b) ) )
```

Comment:
Compilation fails here when the current compiler version is different from the one used internally (currently 406). It seems we need `to_current.copy_directive_argument`.

Reply:
Fixed in master now.